multichain-testing/test/xcs-swap-anything/helpers.ts

```diff
@@ -0,0 +1,826 @@
+      destinationChain,
+      issuingChain,
+      issuingDenom,
+    );
+
+    const getDenomAmount = balances => {
+      const balance = balances.find(coin => coin.denom === `ibc/${denom}`);
+      const amount = balance ? Number(balance.amount) : 0;
+      return amount;
+    };
+
+    await retryUntilCondition(
+      () => destinationClient.getAllBalances(destinationAddress),
+      (coins: Coin[]) => {
+        const before = getDenomAmount(destinationBalanceBefore);
+        const current = getDenomAmount(coins);
+        return current > before;
+      },
+      `Transferred tokens not found on destination address: ${destinationAddress}`,
+    );
+    console.log(
+      `Transferred tokens found on destination address: ${destinationAddress}`,
+    );
+
+    return {
+      client: destinationClient,
+      address: destinationAddress,
+      wallet: destinationWallet,
+    };
+  };
+
+  const invokeOsmosisContract = async (contractAddress: string, txMsg) => {
+    const clientRegistry = osmosisClient.registry;
+    clientRegistry.register(MsgExecuteContract.typeUrl, MsgExecuteContract);
+
+    const fee = {
+      amount: [{ denom: 'uosmo', amount: '1000' }],
+      gas: '200000',
+    };
+
+    const message = MsgExecuteContract.fromPartial({
+      sender: osmosisAddress,
+      contract: contractAddress,
+      msg: toUtf8(JSON.stringify(txMsg)),
+      funds: [],
+    });
+
+    const encodeObjects: Array<import('@cosmjs/proto-signing').EncodeObject> = [
+      {
+        typeUrl: '/cosmwasm.wasm.v1.MsgExecuteContract',
+        value: message,
+      },
+    ];
+
+    const response = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      encodeObjects,
+      fee,
+    );
+    console.log('invokeOsmosisContract DeliverTxResponse: ', response);
+
+    const { msgResponses, code } = response;
+
+    if (code !== 0) {
+      throw Error(`Failed to execute osmosis contract with message ${message}`);
+    }
+
+    return msgResponses;
+  };
+
+  const queryOsmosisContract = async (contractAddress: string, queryMsg) => {
+    const wasmClient = await CosmWasmClient.connect(osmosisRpcEndpoint);
+    const result = await wasmClient.queryContractSmart(
+      contractAddress,
+      queryMsg,
+    );
+
+    return result;
+  };
+
+  const downloadXcsContracts = async (
+    xcsContracts: Contracts,
+    branch: string = osmosisBranch,
+    scriptPath: string = scriptLocalPath,
+    artifactsPath: string = xcsArtifactsPath,
+  ) => {
+    const wasmFiles: string[] = [];
+    for (const contract in xcsContracts) {
+      if (Object.prototype.hasOwnProperty.call(xcsContracts, contract)) {
+        const { label } = xcsContracts[contract];
+        wasmFiles.push(`${label}.wasm`);
+      }
+    }
+
+    try {
+      await execa('bash', [
+        scriptPath,
+        'osmosis-labs',
+        'osmosis',
+        branch,
+        'tests/ibc-hooks/bytecode',
+        artifactsPath,
+        ...wasmFiles,
+      ]);
```

Comment:
If we're going to use `execa`, we might as well take advantage of it:

```suggestion
      await $`bash ${scriptPath} osmosis-labs osmosis ${branch} tests/ibc-hooks/bytecode ${artifactsPath} ${wasmFiles}`;
```

In fact, the scripts should be executable without the extra `bash`, right?

```suggestion
      await $`${scriptPath} osmosis-labs osmosis ${branch} tests/ibc-hooks/bytecode ${artifactsPath} ${wasmFiles}`;
```

Reply:
Changes introduced in commit : e487bb1d7487aa1964323a6bf95cb83e8641e847